src/main/java/com/epam/grid/engine/provider/utils/DirectoryPathUtils.java

```
@@ -0,0 +1,111 @@
+public final class DirectoryPathUtils {
+    private static final String FORWARDSLASH = "/";
+    private static final String ALL_PERMISSIONS_STRING = "rwxrwxrwx";
+
+    /**
+     * If adjustable directory has absolute path, checks, that it begins with gridEngineFolder and exists, creates if
+     * needed.
+     * If adjustable directory has relative path, adds gridEngineFolder to its beginning, checks its existence and
+     * creates, if needed.
+     *
+     * @param adjustableFolder Directory, which should be checked for correction path
+     * @param gridEngineFolder Primary working directory from properties
+     * @return Adjusted directory path with added primary directory added if needed
+     */
+    public static String buildProperDir(final String adjustableFolder, final String gridEngineFolder) {
+        final StringBuilder properDir = new StringBuilder();
+        final File adjustableFolderPath = new File(adjustableFolder);
+        if (adjustableFolderPath.isAbsolute()) {
+            if (!adjustableFolder.startsWith(gridEngineFolder)) {
+                final String errorMessage = "Provided path is absolute, but doesn't begin with Grid Engine folder";
+                log.error(errorMessage);
+                throw new IllegalStateException(errorMessage);
+            }
+            if (!adjustableFolderPath.exists()) {
+                try {
+                    if (adjustableFolderPath.mkdirs()) {
+                        log.info("Directory with path " + adjustableFolderPath + " was created.");
+                        grantAllPermissionsToFolder(adjustableFolderPath);
+                    } else {
+                        final String errorMessage = "Failed to create directory with path " + adjustableFolderPath;
+                        log.error(errorMessage);
+                        throw new IOException(errorMessage);
+                    }
+                } catch (final Exception exception) {
```

I don't think we should swallow exception here, just pass it

---

fixed